### PR TITLE
Add T5Tokenizer based on SentencePieceTokenizer

### PR DESCRIPTION
**Pull request by swapnull7**
Adds a `T5Tokenizer` that will be used with pre-trained T5 model.

### CODE REVIEW

File: texar/torch/data/tokenizers/t5_tokenizer.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+            Unknown token.
+
+        `"sep_token"`: str
+            Separation token.
+
+        `"pad_token"`: str
+            Padding token.
+
+        `"extra_ids"`: int
+                Add a number of extra ids added to the end of the vocabulary
```
Comment (huzecong) [inline texar/torch/data/tokenizers/t5_tokenizer.py]: The indentation here is a bit weird.

File: texar/torch/data/tokenizers/t5_tokenizer_test.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
+"""
+Unit tests for SentencePiece tokenizer.
```
Comment (huzecong) [inline texar/torch/data/tokenizers/t5_tokenizer_test.py]: "for T5 tokenizer"

Comment (swapnull7) [reply texar/torch/data/tokenizers/t5_tokenizer_test.py]: 👍

File: texar/torch/data/tokenizers/t5_tokenizer.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+                 cache_dir: Optional[str] = None,
+                 hparams=None):
+
+        self.load_pretrained_config(pretrained_model_name, cache_dir, hparams)
+
+        if self.pretrained_model_dir is not None:
+            assert self.pretrained_model_name is not None
+            vocab_file = os.path.join(self.pretrained_model_dir,
+                                      self._VOCAB_FILE_NAMES['vocab_file'])
+            assert self.pretrained_model_name is not None
```
Comment (gpengzhi) [inline texar/torch/data/tokenizers/t5_tokenizer.py]: why do we check it twice?

File: texar/torch/data/tokenizers/t5_tokenizer.py [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+                ["<extra_id_{}>".format(i) for i in range(extra_ids)])
+
+        setattr(self.hparams, 'additional_special_tokens',
+                additional_special_tokens)
+
+        super().__init__(hparams=None)
+
+    @staticmethod
+    def default_hparams() -> Dict[str, Any]:
+        r"""Returns a dictionary of hyperparameters with default values.
```
[the remark this answers is not in the document]
Comment (swapnull7) [reply texar/torch/data/tokenizers/t5_tokenizer.py]: Fixed.

File: texar/torch/data/tokenizers/t5_tokenizer.py [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+            'pretrained_model_name': 'T5-Small',
+            'vocab_file': None,
+            'max_len': 512,
+            'bos_token': None,
+            'eos_token': '</s>',
+            'unk_token': '<unk>',
+            'pad_token': '<pad>',
+            'extra_ids': 100,
+            'additional_special_tokens': [],
+            'do_lower_case': False,
```
[the remark this answers is not in the document]
Comment (swapnull7) [reply texar/torch/data/tokenizers/t5_tokenizer.py]: We don't removed this from the default params

File: texar/torch/data/tokenizers/t5_tokenizer.py (hunk 3 above)
Comment (swapnull7) [reply texar/torch/data/tokenizers/t5_tokenizer.py]: We should'nt. Removing it from `BERTTokenizer` too.

**Pull request merged**